chromium_src/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc

```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "build/build_config.h"
+#include "build/chromeos_buildflags.h"
+
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_CHROMEOS_LACROS)
+#define CHROME_BROWSER_WEB_APPLICATIONS_COMMANDS_CLEAR_BROWSING_DATA_COMMAND_H_
```

Comment:
I'm confused by this, why are we changing upstream behavior here?

Reply:
The error it fixes is
```
In file included from ../../../src/chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc:75:
In file included from ../../chrome/browser/web_applications/commands/clear_browsing_data_command.h:9:
In file included from ../../chrome/browser/web_applications/web_app_provider.h:14:
In file included from ../../chrome/browser/web_applications/externally_managed_app_manager.h:17:
In file included from ../../chrome/browser/web_applications/external_install_options.h:16:
In file included from ../../chrome/browser/web_applications/web_app_install_params.h:15:
In file included from ../../chrome/browser/web_applications/web_app_install_utils.h:11:
In file included from ../../chrome/browser/web_applications/os_integration/os_integration_manager.h:19:
In file included from ../../chrome/browser/web_applications/os_integration/url_handler_manager.h:14:
../../chrome/browser/web_applications/web_app_origin_association_manager.h:14:10: fatal error: 'components/webapps/services/web_app_origin_association/public/mojom/web_app_origin_association_parser.mojom.h' file not found
#include "components/webapps/services/web_app_origin_association/public/mojom/web_app_origin_association_parser.mojom.h"
```

Since that file is not supposed to be used on Android I simply excluded those headers. I'm not sure how it's handled in the upstream. Do you see better solution for this error?

Reply:
@samartnik which file is not supposed to be used on android?

Reply:
missing not generated file `web_app_origin_association_parser.mojom.h`

Reply:
sorry, I meant those 3 excluded are not supposed to be used on Android

Reply:
so this looks like an upstream bug because https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc;l=75-78;drc=359c4471ef9ffdf94d1080815d51f28ce0302b94;bpv=1;bpt=1 should be in a `!IS_ANDROID` guard. 
This should have explanation in comments and we can remove `IS_CHROMEOS_LACROS` and remove this entire thing when upstream fixes it

Reply:
have we filed an upstream issue for it?

Reply:
added comment and removed `IS_CHROMEOS_LACROS` https://github.com/brave/brave-core/commit/95edc727e9cef1594c2b1f9692c0338c5e41cba6

Reply:
We didn't file issue, I'm sure how to properly do it. Do we have wiki on that?

Reply:
Created https://bugs.chromium.org/p/chromium/issues/detail?id=1305046